akka-diagnostics/src/main/scala/akka/diagnostics/ConfigChecker.scala

```diff
@@ -0,0 +1,1169 @@
+        if (config.getBoolean("akka.diagnostics.checker.fail-on-warning")) FailOnWarnings
+        else LogWarnings
+      } else Disabled
+
+    sealed trait Mode
+    case object Disabled extends Mode
+    case object LogWarnings extends Mode
+    case object FailOnWarnings extends Mode
+  }
+
+  private implicit class ConfigTools(val c: Config) extends AnyVal {
+    def stringValue(path: String): Option[String] =
+      if (c.hasPath(path)) {
+        val value = c.getValue(path)
+        if (value.valueType() == ConfigValueType.STRING) Some(value.unwrapped().asInstanceOf[String])
+        else None
+      } else None
+
+    def numericValue(path: String): Option[Int] =
+      if (c.hasPath(path)) {
+        val value = c.getValue(path)
+        if (value.valueType() == ConfigValueType.NUMBER) Some(value.unwrapped().asInstanceOf[Int])
+        else None
+      } else None
+
+    def hasPathOfType(path: String, valueType: ConfigValueType): Boolean =
+      c.hasPath(path) && c.getValue(path).valueType() == valueType
+  }
+}
+
+/**
+ * The `ConfigChecker will try to find potential configuration issues. It is run when the actor system is started. It
+ * also possible to run it as a Java main program, see [[ConfigChecker#main]].
+ *
+ * Detailed documentation can be found in the `akka.diagnostics.checker` section of the reference.conf and in the
+ * "Configuration Checker" section of the Akka Reference Documentation.
+ */
+class ConfigChecker(system: ExtendedActorSystem, config: Config, reference: Config) {
+  import ConfigChecker._
+
+  def this(system: ExtendedActorSystem) =
+    this(system, system.settings.config, ConfigFactory.defaultReference(system.dynamicAccess.classLoader))
+
+  private val disabledChecks: Set[String] =
+    config.getStringList("akka.diagnostics.checker.disabled-checks").asScala.toSet
+
+  private val confirmedPowerUserSettings: Set[String] =
+    config.getStringList("akka.diagnostics.checker.confirmed-power-user-settings").asScala.toSet
+  private[akka] val (powerUserSettings: Set[String], powerUserWildcardSettings: Set[String]) = {
+    val fullList =
+      expandPowerUserSettings(config.getStringList("akka.diagnostics.checker.power-user-settings").asScala.toSet)
+        .diff(confirmedPowerUserSettings)
+    val (wildcards, exact) = fullList.partition(_.endsWith(".*"))
+    val wildcardPrefixes = wildcards.map(s => s.substring(0, s.length - 2))
+
+    (exact, wildcardPrefixes)
+  }
+
+  private def expandPowerUserSettings(settings: Set[String]): Set[String] = {
+//    // FIXME remove these automatic additions when only using AKka 2.6
+//    if (isAkka27)
+//      settings.flatMap { s =>
+//        if (s.startsWith("akka.remote.") && !s.startsWith("akka.remote.classic.") && !s.startsWith("akka.remote.artery."))
+//          s :: remoteConfigPath(s) :: Nil // add .classic automatically for Akka 2.5 and 2.6 support
+//        else
+//          s :: Nil
+//      }
+//    else
```

Comment:
we can remove this, and change the reference.conf to the property names that are used in Akka 2.7

Reply:
removed.
WIP find the new property names from 2.7

Reply:
Done